projects/components/src/splitter/splitter.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable } from '@angular/core';
+import { fromEvent, Observable } from 'rxjs';
+import { map, mergeMap, takeUntil } from 'rxjs/operators';
+import { SplitterDirection } from './splitter';
+
+@Injectable()
+export class SplitterService {
+  public buildSplitterLayoutChangeObservable(
+    splitterElement: HTMLElement,
+    parentElement: HTMLElement,
+    direction: SplitterDirection
+  ): Observable<boolean> {
+    const moveOnParent$ = fromEvent<MouseEvent>(parentElement, 'mousemove');
+    const downOnHost$ = fromEvent<MouseEvent>(splitterElement, 'mousedown');
+    const upOnWindow$ = fromEvent<MouseEvent>(window, 'mouseup');
+
+    return downOnHost$.pipe(
+      mergeMap(() => moveOnParent$.pipe(takeUntil(upOnWindow$))),
+      map(event => {
+        const parentBounds = parentElement.getBoundingClientRect();
+        const prevSibling = this.getHTMLElement(splitterElement.previousElementSibling);
+        const nextSibling = this.getHTMLElement(splitterElement.nextElementSibling);
+
+        const previousElementUpdatedDimension = prevSibling
+          ? this.calculateUpdatedDimensionForPreviousElement(event, parentBounds, direction)
+          : 0;
+        const nextElementUpdatedDimension = nextSibling
+          ? this.calculateUpdatedDimensionForNextElement(previousElementUpdatedDimension, parentBounds, direction)
+          : 0;
+        const updatedDimensionData: UpdatedDimensionData[] = [];
+
+        if (prevSibling) {
+          updatedDimensionData.push({
+            element: prevSibling,
+            dimension: previousElementUpdatedDimension
+          });
+        }
+
+        if (nextSibling) {
+          updatedDimensionData.push({
+            element: nextSibling,
+            dimension: nextElementUpdatedDimension
+          });
+        }
+
+        return updatedDimensionData;
+      }),
+      map(data => this.maybeUpdateLayoutOfSiblings(data, direction))
+    );
+  }
+
+  private maybeUpdateLayoutOfSiblings(data: UpdatedDimensionData[], direction: SplitterDirection): boolean {
+    let layoutChanged = false;
+    data.forEach(datum => {
+      if (direction === SplitterDirection.Horizontal) {
+        this.updateWidth(datum.element, datum.dimension);
```

Comment:
this seems inverted to me. I'd expect a horizontal splitter updates height, and a vertical width. In other words, the orientation describes the splitter itself, not how it affects the dimensions of its siblings.

Reply:
Okay, so i added a `direction` property which represents the direction in which the splitter could move. It does not represent the orientation of splitter. If this is causing confusion then I could rename the property.

Reply:
It's the same purpose though right? When describing a splitter as vertical, I'd expect it to render... vertically. Kind of example: https://material.angular.io/components/divider/overview#vertical-divider

Reply:
Yeah, my take was that the direction it can move in is more important. But i can see it is confusion. Will update.